packages/kotti-ui/source/kotti-filter/components/FilterRow.vue

```diff
@@ -121,24 +133,14 @@ export default defineComponent<{
 			() => operationOptions.value.length <= 1,
 		)
 
-		const getPrefix = (index: number): string =>
-			index === 0 ? translations.value.whereLabel : translations.value.andLabel
-		const handleRemoveFilter = () => emit('removeFilter', props.filter.key)
-
-		watch(filterData, (filterData, prevFilterData) => {
-			if (
-				filterData.key === prevFilterData.key &&
-				filterData.operation === prevFilterData.operation &&
-				filterData.value === prevFilterData.value
-			)
-				return
-			emit('setFilter', filterData, props.index)
-		})
+		const handleRemove = () => emit('remove', props.filterKey)
```

Comment:
I don’t think emitting `props.filterKey` is necessary. The parent should already know what it belongs to